pkg/reconciler/revision/resolve_test.go

```diff
@@ -55,10 +55,13 @@ func mustDigest(t *testing.T, img v1.Image) v1.Hash {
 	return h
 }
 
-func fakeRegistry(t *testing.T, repo, username, password string, img v1.Image) *httptest.Server {
+func fakeRegistry(t *testing.T, repo, username, password, ua string, img v1.Image) *httptest.Server {
 	manifestPath := fmt.Sprintf("/v2/%s/manifests/latest", repo)
 	const basicAuth = "Basic "
 	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if got, want := r.Header.Get("User-Agent"), ua; !strings.Contains(got, want) {
```

Comment:
wouldn't this be a `==` match?

Reply:
The string you pass in gets annotated with go-containerregistry version information (which is why this is more useful than just setting `User-Agent` in the transport).

Reply:
I see.